src/query/plan/operator.cpp

```diff
@@ -2564,39 +2564,63 @@ void Delete::DeleteCursor::UpdateDeleteBuffer(Frame &frame, ExecutionContext &co
     expression_results.emplace_back(expression->Accept(evaluator));
   }
 
+  auto vertex_auth_checker = [&context](const VertexAccessor &va) {
+#ifdef MG_ENTERPRISE
+    if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
+        !context.auth_checker->Has(va, storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE)) {
+      throw QueryRuntimeException("Vertex not deleted due to not having enough permission!");
+    }
+#endif
+  };
+
+  auto edge_auth_checker = [&context](const EdgeAccessor &ea) {
+#ifdef MG_ENTERPRISE
+    if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
+        !(context.auth_checker->Has(ea, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE) &&
+          context.auth_checker->Has(ea.To(), storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::UPDATE) &&
+          context.auth_checker->Has(ea.From(), storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::UPDATE))) {
+      throw QueryRuntimeException("Edge not deleted due to not having enough permission!");
+    }
+#endif
+  };
+
   for (TypedValue &expression_result : expression_results) {
     AbortCheck(context);
     switch (expression_result.type()) {
       case TypedValue::Type::Vertex: {
         auto va = expression_result.ValueVertex();
-#ifdef MG_ENTERPRISE
-        if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
-            !context.auth_checker->Has(va, storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE)) {
-          throw QueryRuntimeException("Vertex not deleted due to not having enough permission!");
-        }
-#endif
+        vertex_auth_checker(va);
         buffer_.nodes.push_back(va);
         break;
       }
       case TypedValue::Type::Edge: {
         auto ea = expression_result.ValueEdge();
-#ifdef MG_ENTERPRISE
-        if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
-            !(context.auth_checker->Has(ea, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE) &&
-              context.auth_checker->Has(ea.To(), storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::UPDATE) &&
-              context.auth_checker->Has(ea.From(), storage::View::NEW,
-                                        query::AuthQuery::FineGrainedPrivilege::UPDATE))) {
-          throw QueryRuntimeException("Edge not deleted due to not having enough permission!");
-        }
-#endif
+        edge_auth_checker(ea);
         buffer_.edges.push_back(ea);
         break;
       }
+      case TypedValue::Type::Path: {
+        auto path = expression_result.ValuePath();
+#ifdef MG_ENTERPRISE
+        std::transform(path.edges().cbegin(), path.edges().cend(), std::back_inserter(buffer_.edges),
+                       [&edge_auth_checker](const auto &ea) {
+                         edge_auth_checker(ea);
+                         return ea;
+                       });
```

Comment:
Splitting in two should be faster and a better control flow.
Is there a reason why you prefer `std::transform`?

Reply:
tests we have in openCypher gql tests. I will just move them to main gql file. 
This task is very "functional" and corresponds to filter operation in functional programming. Process collection and take only those which satisfy filter.

Reply:
But you are throwing if auth fails. I assume somewhere above we catch, but in any case, you will terminate the `std::transform` on first throw, you will not copy just the authorized ones.

Plus, I am not sure the correct approach would be to delete just the elements we are authorized to. I would rather fail to delete anything if are not authorized to delete any one of the path's elements.

Reply:
oh yes good point